packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamTable.tsx

```diff
@@ -1,89 +1,62 @@
+import { BackendDestinationEnum, getEndpoint } from '../../utils/functions'
 
 export type ChallengeTeamTableProps = {
   challengeId: string
-  onSelectTeam: (team: Team) => void
+  selectedTeamId?: string
+  onSelectTeam: (teamId?: string) => void
 }
 
+// Custom type for the challenge team table that combines fields from the Team and the ChallengeTeam record
 type ChallengeTeamRow = {
   id: string
   name: string
   created: string
   description?: string
 }
 
-export default function ChallengeTeamTable({
-  challengeId,
-  onSelectTeam,
-}: ChallengeTeamTableProps) {
-  const [allRows, setAllRows] = useState<ChallengeTeamRow[]>([])
-  const [teamIdList, setTeamIdList] = useState<string[]>([])
-  const [teamsById, setTeamsById] = useState<Record<string, Team>>({})
-  const { data: regTeams } = useGetChallengeTeamList(challengeId)
-
-  const [selectedTeam, setSelectedTeam] = useState<
-    string | number | undefined
-  >()
-
-  const { data: teamsList, isLoading } = useGetTeamList(teamIdList, {
-    enabled: !!teamIdList.length,
-  })
-
-  const teamChangeHandler = (value: string | number) => {
-    setSelectedTeam(value)
-    onSelectTeam(teamsById[value])
-  }
-
-  const getTeamRow = (team: Team): ChallengeTeamRow => {
-    return {
-      id: team.id,
-      name: team.name,
-      created: formatDate(dayjs(team.createdOn), 'MM/DD/YY'),
-      description: team.description,
-    }
+const getTeamRow = (team: Team): ChallengeTeamRow => {
+  return {
+    id: team.id,
+    name: team.name,
+    created: formatDate(dayjs(team.createdOn), 'MM/DD/YY'),
+    description: team.description,
   }
+}
 
-  useEffect(() => {
-    const ids = regTeams?.map(team => team.teamId) ?? []
-    setTeamIdList(ids)
-  }, [regTeams])
+export default function ChallengeTeamTable(props: ChallengeTeamTableProps) {
+  const { challengeId, onSelectTeam, selectedTeamId } = props
+  const { data: registeredTeams, isLoading: isLoadingChallengeTeams } =
+    useGetChallengeTeamList(challengeId)
+  const [searchValue, setSearchValue] = useState('')
+  const teamIdList = registeredTeams?.map(team => team.teamId) ?? []
+  const { data: teamsList, isLoading: isLoadingTeamList } = useGetTeamList(
+    teamIdList,
+    {
+      enabled: !!teamIdList.length,
+    },
+  )
 
-  useEffect(() => {
+  const allRows: ChallengeTeamRow[] = useMemo(() => {
+    const term = searchValue.toLowerCase()
     const teams =
-      teamsList?.list.filter(team => team.canRequestMembership !== false) ?? []
-    const row: ChallengeTeamRow[] = []
-    const teamRecords: Record<string, Team> = {}
-    teams.forEach(team => {
-      row.push(getTeamRow(team))
-      teamRecords[team.id] = team
-    })
-    setAllRows(row)
-    setTeamsById(teamRecords)
-  }, [teamsList])
+      teamsList?.list.filter(
+        team =>
+          team.canPublicJoin ||
+          // `canRequestMembership` is true by default if the property is not present
+          team.canRequestMembership !== false,
```

Comment:
This broke when I tried to simplify it by assuming `canRequestMembership` defaulted to `false`, when it actually defaults to `true`! This was confusing, so I added a comment here and filed PLFM-8304